File: R/Friday Purchases.R
library(dplyr)
library(readxl)
library(lubridate)
df <- read_excel('E:\\Data_analysis_projects\\Friday Purchases\\dataset\\Friday Purchases.xlsx')
fridays_q1 <- df %>%
  filter(day_name == 'Friday' & month(date) <= 3)

fridays_q1 <- fridays_q1 %>%
  mutate(week_number = isoweek(date)) %>%
  group_by(week_number) %>%
  summarise(average_amount_spent = mean(amount_spent))

print(fridays_q1)

File: python/Friday Purchases.py
import pandas as pd

# Load the data
df = pd.read_excel(r'E:\\Data_analysis_projects\\Friday Purchases\\dataset\\Friday Purchases.xlsx')

df['week_num'] = df['date'].dt.isocalendar().week

df1 = df[(df['date'].dt.month <= 3) & (df['day_name'] == 'Friday')]

df2 = df1.groupby('week_num')['amount_spent'].mean().reset_index()

print(df2.sort_values(by='amount_spent',ascending=False))
